service/service.go

```diff
@@ -190,15 +210,18 @@ func (app *Application) runAndWaitForShutdownEvent() {
 	}
 }
 
-func (app *Application) setupConfigurationComponents() {
+func (app *Application) setupConfigurationComponents(cFactory ConfigFactory) {
 	// Load configuration.
 	app.logger.Info("Loading configuration...")
-	cfg, err := config.Load(app.v, app.factories, app.logger)
+	cfg, err := cFactory(app.v, app.factories)
 	if err != nil {
 		log.Fatalf("Cannot load configuration: %v", err)
 	}
-
 	app.config = cfg
+	err = config.ValidateConfig(app.config, app.logger)
+	if err != nil {
+		log.Fatalf("Cannot load configuration: %v", err)
```

Comment:
Note that #627 eliminates the usage of `log.Fatal` in `Application`, so we may need to do a follow-up cleanup on this code.

Reply:
If #627 gets in first I will rebase. If this one gets first #627 will have to rebase :)